File: ast/ast_types/name.go
package ast_types

import "github.com/panda-io/micro-panda/ast/ast"

type TypeName struct {
	TypeBase
	Name      string
	Qualified string

	IsEnum bool
}

func (n *TypeName) Equal(t ast.Type) bool {
	if t, ok := t.(*TypeName); ok {
		return n.Qualified != "" && t.Qualified == n.Qualified
	}
	return false
}

func (n *TypeName) String() string {
	if n.Qualified != "" {
		return n.Qualified
	}
	return n.Name
}

File: app/src/main/java/com/github/panda_io/micro_panda/ast/type/Name.java
package com.github.panda_io.micro_panda.ast.type;

public class Name extends Type {
    String name;

    public String qualified;
    public boolean isEnum;

    public Name(String name) {
        this.name = name;
    }

    public boolean equal(Type type) {
        if (type instanceof Name) {
            return this.qualified != null && this.qualified.equals(((Name) type).qualified);
        }
        return false;
    }

    public String string() {
        if (this.qualified != null) {
            return this.qualified;
        }
        return this.name;
    }
}
